src/protocols/secure_channel/tests/TestCheckinMsg.cpp

```diff
@@ -102,20 +118,24 @@ void TestCheckin_Generate(nlTestSuite * inSuite, void * inContext)
         // ChipLogError(Inet, "%s", err.AsString());
         // NL_TEST_ASSERT(inSuite, (CHIP_NO_ERROR == err));
 
+        // Testing empty application data
         ByteSpan emptyData;
-        err = CheckinMessage::GenerateCheckinMessagePayload(keyHandle, counter, emptyData, outputBuffer);
+        err = CheckinMessage::GenerateCheckinMessagePayload(aes128KeyHandle, hmac128KeyHandle, counter, emptyData, outputBuffer);
         NL_TEST_ASSERT(inSuite, (CHIP_NO_ERROR == err));
 
+        // Testing empty output buffer
         MutableByteSpan empty;
-        err = CheckinMessage::GenerateCheckinMessagePayload(keyHandle, counter, emptyData, empty);
+        err = CheckinMessage::GenerateCheckinMessagePayload(aes128KeyHandle, hmac128KeyHandle, counter, emptyData, empty);
         NL_TEST_ASSERT(inSuite, (CHIP_ERROR_INVALID_ARGUMENT == err));
 
+        // Test incorrect output buffer size
         userData = chip::ByteSpan(gargantuaBuffer, sizeof(gargantuaBuffer));
-        err      = CheckinMessage::GenerateCheckinMessagePayload(keyHandle, counter, userData, outputBuffer);
+        err = CheckinMessage::GenerateCheckinMessagePayload(aes128KeyHandle, hmac128KeyHandle, counter, userData, outputBuffer);
```

Comment:
It seems you are testing for an output buffer that is too small by adding an input buffer that is too big. Either fix the comment on that or maybe test both cases if they differ enough?

Reply:
Updated comment